### PR TITLE
[HttpClient] Remove OTEL_SEMCONV_STABILITY_OPT_IN

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM.

Do you want to remove the unused http client based entries from `SemanticConventions.cs`?

**Comment by vishweshbankwar** [comment]
> LGTM.
> 
> Do you want to remove the unused http client based entries from `SemanticConventions.cs`?

@rajkumar-rangaraj Will do separately: https://github.com/open-telemetry/opentelemetry-dotnet/issues/5071

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs [hunk 1]
```diff
@@ -275,19 +250,11 @@ public void OnStopActivity(Activity activity, object payload)
                     activity.SetStatus(SpanHelper.ResolveSpanStatusForHttpStatusCode(activity.Kind, (int)response.StatusCode));
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs:250]: Could you hardcode the kind to `ActivityKind.Client` here as well similar to how you have done it for the webrequest one?

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs (hunk 1 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs:250]: in web request we cannot rely on activity being non-null always. here we can, but I can update this one as well later.

**Comment by vishweshbankwar** [comment]
> @utpilla @rajkumar-rangaraj @vishweshbankwar Most of vendors' exporter in OTel Collector are still not support translate new http attributes, could you help understand how do we avoid the breaking if no environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` support ?

@wangzlei Please refer to FAQ of [this announcement](https://github.com/open-telemetry/opentelemetry-dotnet/discussions/5179)

**Pull request merged**